azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NetworkInterface.java

```diff
@@ -81,35 +81,42 @@
      * @throws CloudException exceptions thrown from the cloud.
      * @throws IOException exceptions thrown from serialization/deserialization.
      */
-    PublicIpAddress publicIpAddress() throws CloudException, IOException;
+    PublicIpAddress primaryPublicIpAddress() throws CloudException, IOException;
 
     /**
      * @return the resource id of the virtual network subnet associated with this network interface.
      */
-    String subnetId();
+    String primarySubnetId();
```

Comment:
BTW: TODO won't pass checkstyle (just learned the hard way :-) so just log an issue and assign to self